src/cudamatrix/cu-vector.cc

```diff
@@ -711,6 +711,47 @@ void CuVectorBase<Real>::MulElements(const CuVectorBase<Real> &v) {
 
 template<>
 template<>
+void CuVectorBase<double>::CopyFromVec(const CuVectorBase<double> &src) {
+  KALDI_ASSERT(src.Dim() == dim_);
+#if HAVE_CUDA == 1
+  if (CuDevice::Instantiate().Enabled()) {
+    if (dim_ == 0) return;
+    Timer tim;
+    int dimBlock(CU2DBLOCK);
+    int dimGrid(n_blocks(dim_, CU2DBLOCK));
+    cuda_copy_from_vec(dimGrid, dimBlock, data_, src.data_, dim_);
+    CU_SAFE_CALL(cudaGetLastError());
+    CuDevice::Instantiate().AccuProfile(__func__, tim.Elapsed());
+  } else
+#endif
+  {
+    Vec().CopyFromVec(src.Vec());
+  }
+}
+
+template<>
+template<>
+void CuVectorBase<float>::CopyFromVec(const CuVectorBase<float> &src) {
```

Comment:
I think you don't need this float to float and double to double version of CopyFromVec and the vectors with same type is handled in the main template version of function!